### DISCUSSION TITLE
COLUMNS() regex to optionally include table name

### DISCUSSION THREAD

**Discussion by stevenpelley** [duckdb/duckdb#12139] [Ideas]
Use case: I'm performing a self join and would like to name/alias the resulting expressions with their original columns names prefixed with the side of the join.  [COLUMNS() regex](https://duckdb.org/docs/sql/expressions/star#columns-regular-expression) allows filtering and renaming columns but does not include the table/relation and so cannot easily include the join side or filter based on this.  I see that https://github.com/duckdb/duckdb/pull/7499 specifically removed including the table/relation from COLUMNS() lambda (and so I assume also regex); this motivation makes sense and should remain the default behavior.  I'm asking to be able to change the behavior using an optional argument to COLUMNS() or using a PRAGMA/SET.

An example:
```
create temp table t (c int);
insert into t values (0), (1), (2), (10), (11);
select * from t;
c 
--
0 
1 
2 
10
11

-- the result I'd like
SELECT t1.c AS left_c, t2.c AS right_c FROM t AS t1 ASOF JOIN t AS t2 ON t2.c < t1.c ORDER BY t1.c;
left_c  right_c
------  -------
1       0      
2       1      
10      2      
11      10

-- they query I want to write
SELECT COLUMNS('(.*)\\.(.*)', true) AS '\\1_\\2' FROM t AS t1 ASOF JOIN t AS t2 ON t2.c < t1.c ORDER BY t1.c;
```

Right now I can rely on duplicate identifiers being suffixed with '_1' to some extent but it gets confusing.  In the above example the table has only 1 column but in cases with many columns, and where I might add additional columns in the future, I don't want to have to modify the SELECT expressions and list expressions individually.

**Comment by Tishj** [comment]
I might think it's fine to add it as an optional keyword argument, I'm not a fan of the `COLUMNS(..., true)` syntax

**Comment by soerenwolfers** [comment]
What about something like tablename.COLUMNS(*)? Adding the table name just to then filter on it feels awkward

**Reply by Tishj** [reply]
That would mean columns needs to be reserved
```
D create table tbl(a varchar);
D create schema tbl;
D create macro tbl.columns() as 42;
D select tbl.columns(*) from tbl;
┌───────────────┐
│ tbl.columns() │
│     int32     │
├───────────────┤
│    0 rows     │
└───────────────┘
```
